README.md

```diff
@@ -1,102 +1,262 @@
+
+#### Options
+- `entity_id` - **Required** - Unique identifier for the service provider, often the url of the metadata file.
+- `private_key` - **Required** - Private key (PEM format string) for the service provider.
+- `certificate` - **Required** - Certificate (PEM format string) for the service provider.
+- `assert_endpoint` - **Required** - URL of service provider assert endpoint.
+- `force_authn` - Boolean that allows forcing re-authentication of users even if the user has a SSO session with the IdP.  This can also be configured on the [IdP](#IdentityProvider) or on a per-method basis.
+- `auth_context` - Specifies `AuthnContextClassRef`.  This can also be configured per-method basis.
+- `nameid_format` - Format for Name ID.  This can also be configured on a per-method basis.
+- `sign_get_request` -  Boolean, if true signs the request.  This can also be configured on the [IdP](#IdentityProvider) or on a per-method basis.
+- `allow_unencrypted_assertion` -  Boolean, when set to true allows unencrypted assertions.  This can also be configured on the [IdP](#IdentityProvider) or on a per-method basis.
 
-Upon creating at least one service provider and one identity provider, you can then create SAML requests between them.
+#### Returns the following functions
+- [`create_login_request_url`](#create_login_request_url) - Get a URL to initiate a login.
+- [`redirect_assert`](#redirect_assert) - Gets a user object if the login attempt is valid, inflates the response before parsing it.
+- [`post_assert`](#post_assert) - Gets a user object if the login attempt is valid.
+- [`create_logout_request_url`](#create_logout_request_url)- Creates a request URL to log out the user.
+- [`create_logout_response_url`](#create_logout_response_url) - Creates a URL to confirm a successful logout.
+- [`create_metadata`](#create_metadata) - Returns the XML metadata used during the initial SAML configuration.
 
-```coffee
-  # -- REQUIRED --
-  # Returns a redirect URL, at which a user can login
-  sp.create_login_url(idp, cb)
+#### Example
+```javascript
 
-  # Returns user object, if the login attempt was valid.
-  sp.assert(idp, request_body, cb)
+  // Initialize options object
+  var sp_options = {
+    entity_id: "https://sp.example.com/metadata.xml",
+    private_key: fs.readFileSync("key-file.pem").toString(),
+    certificate: fs.readFileSync("cert-file.crt").toString(),
+    assert_endpoint: "https://sp.example.com/assert",
+    force_authn: true,
+    auth_context: { comparison: "exact", class_refs: ["urn:oasis:names:tc:SAML:1.0:am:password"] },
+    nameid_format: "urn:oasis:names:tc:SAML:2.0:nameid-format:transient",
+    sign_get_request: false,
+    allow_unencrypted_assertion: true
+  }
 
-  # -- OPTIONAL --
-  # Returns a redirect URL, at which a user is logged out.
-  sp.create_logout_url(idp, cb)
+  // Call service provider constructor with options
+  var sp = saml2.ServiceProvider(sp_options);
```

Comment:
Do you not have to do `sp = new saml2.ServiceProvider(sp_options)`? If not these probably aren't constructors, but rather something else.